src/awkward/contents/indexedoptionarray.py

```diff
@@ -1767,6 +1768,22 @@ def _is_equal_to(
             )
         )
 
+    def _trim(self) -> Self:
+        if self._index.length == 0:
+            return self
+
+        nplike = self._backend.index_nplike
+        idx_buf = nplike.asarray(self._index.data, copy=True)
+        only_positive = idx_buf >= 0
+        min_idx = nplike.min(idx_buf[only_positive])
+        max_idx = nplike.max(idx_buf[only_positive])
```

Comment:
@jpivarski should we create a kernel to perform `min_and_max`?

Reply:
To avoid two loops over the data? Maybe that can be a later improvement.

Hmmm. https://stackoverflow.com/q/12200580/1623645

Although that would be an improvement, I don't think it should hold up this PR. Also notice that there are other $\mathcal{O}(n)$ array-traversals in the same function; this won't increase the runtime by a large _fraction_.